analyzers/tests/SonarAnalyzer.UnitTest/Common/AnalyzerLanguageTest.cs

```diff
@@ -26,89 +26,43 @@ namespace SonarAnalyzer.UnitTest.Common
     public class AnalyzerLanguageTest
     {
         [TestMethod]
-        public void Parse()
+        public void ToString_ReturnsLanguageName()
         {
-            AnalyzerLanguage.Parse("cs").Should().Be(AnalyzerLanguage.CSharp);
-            AnalyzerLanguage.Parse("vbnet").Should().Be(AnalyzerLanguage.VisualBasic);
+            AnalyzerLanguage.CSharp.ToString().Should().Be("C#");
+            AnalyzerLanguage.VisualBasic.ToString().Should().Be("Visual Basic");
         }
 
-        [TestMethod]
-        public void Parse_Fail() =>
-            Assert.ThrowsException<NotSupportedException>(() => AnalyzerLanguage.Parse("csharp"));
-
-        [TestMethod]
-        public void GetDirectory()
-        {
-            AnalyzerLanguage.CSharp.DirectoryName.Should().Be("CSharp");
-            AnalyzerLanguage.VisualBasic.DirectoryName.Should().Be("VisualBasic");
-        }
-
-        [TestMethod]
-        public void GetQualityProfileRepositoryKey()
-        {
-            AnalyzerLanguage.CSharp.RepositoryKey.Should().Be("csharpsquid");
-            AnalyzerLanguage.VisualBasic.RepositoryKey.Should().Be("vbnet");
-        }
-
-        [TestMethod]
-        public void AddLanguage()
```

Comment:
I'm glad you're removing this, it's hurts my head to review this and understand why this would have been needed...

Reply:
**AFAICT:**
This has never been needed. It was prepared for a theoretical scenario when single class implements rule for C# and VisualBasic at the same time and is decorated with attributes for both languages.

We've never used this. The rest of the codebase (like RuleFinder) would not digest it nicely anyway 😄